src/plugins/kibana_react/public/field_button/field_button.tsx

```diff
@@ -100,7 +100,16 @@ export function FieldButton({
 
   return (
     <div className={classes} {...rest}>
-      <button onClick={onClick} {...buttonProps} className={buttonClasses}>
+      <button
+        onClick={(e) => {
+          if (e.type === 'click') {
+            e.currentTarget.focus();
+          }
+          onClick();
+        }}
```

Comment:
@andreadelrio @wylieconlon that's how I could solve the flickering of the add button in Discover, by automatically setting focus to sidebar element you've clicked, but of course this also changes focus behavior, but I think that's ok. What do you think?

Reply:
@kertal Found something else
![a_moreFocus](https://user-images.githubusercontent.com/4016496/91483230-36aa7f00-e85c-11ea-9e26-232784519059.gif)


Can we make it so that when I add a field all other fields lose focus? Right now, if I have one field focused (`manufacturer`) and I add a different field (`geoip.continent_name`) the initial field retains its focus state.

Reply:
well, one way to solve this would be to set focus the add/remove button element when it is clicked. Then Chrome doesn't set focus in this case, you can try it out / test it, added this change

Reply:
is this something specific to discover or should we really do this everywhere ? cc @streamich

Reply:
@ppisljar it's also used in Lens